crates/eframe/src/native/run.rs

```diff
@@ -789,190 +1005,666 @@ mod glow_integration {
+                    .take()
+                    .unwrap()
+                    .make_not_current()
+                    .unwrap()
+                    .make_current(window.gl_surface.as_ref().unwrap())
+                    .unwrap(),
+            );
+
+            if !window
+                .gl_surface
+                .as_ref()
+                .unwrap()
+                .is_current(glutin.current_gl_context.as_ref().unwrap())
+            {
+                let builder = &&glutin.builders[&window.id_pair.this];
+                log::error!("egui::create_viewport_sync with title: `{:?}` is not created in main thread, try to use wgpu!", builder.title.clone().unwrap_or_default());
+            }
+
+            egui_glow::painter::clear(gl, screen_size_in_pixels, [0.0, 0.0, 0.0, 0.0]);
+
+            let pixels_per_point = egui_ctx.input_for(id_pair.this, |i| i.pixels_per_point());
+            painter.paint_and_update_textures(
+                screen_size_in_pixels,
+                pixels_per_point,
+                &clipped_primitives,
+                &output.textures_delta,
+            );
+            crate::profile_scope!("swap_buffers");
+            if let Err(err) = window
+                .gl_surface
+                .as_ref()
+                .expect("failed to get surface to swap buffers")
+                .swap_buffers(
+                    glutin
+                        .current_gl_context
+                        .as_ref()
+                        .expect("failed to get current context to swap buffers"),
+                )
+            {
+                log::error!("swap_buffers failed: {err}");
+            }
+            winit_state.handle_platform_output(
+                &win,
+                id_pair.this,
+                egui_ctx,
+                output.platform_output,
+            );
+        }
+
+        fn process_viewport_builders(
+            glutin_ctx: &Rc<RefCell<GlutinWindowContext>>,
+            mut viewports: Vec<ViewportOutput>,
+        ) {
+            let mut active_viewports_ids = vec![ViewportId::ROOT];
+
+            viewports.retain_mut(
+                |ViewportOutput {
+                     builder,
+                     id_pair: ViewportIdPair { this: id, .. },
+                     viewport_ui_cb,
+                 }| {
+                    let mut glutin = glutin_ctx.borrow_mut();
+                    let last_builder = glutin.builders.entry(*id).or_insert(builder.clone());
+                    let (commands, recreate) = changes_between_builders(builder, last_builder);
+                    drop(glutin);
+                    if let Some(viewport) = glutin_ctx.borrow().viewports.get(id) {
+                        let mut viewport = viewport.borrow_mut();
+                        if recreate {
+                            viewport.window = None;
+                            viewport.gl_surface = None;
+                            viewport.id_pair.parent = *id;
```

Comment:
Why is parent changed to point to itself here?

Reply:
That was a mistake!

Reply:
Fixed!